deploy/docker/crawler_pool.py

```
@@ -15,35 +15,48 @@
     try:
-        sig = _sig(cfg)
+        sig = _sig(cfg, crawler_strategy=crawler_strategy)
         async with LOCK:
             if sig in POOL:
-                LAST_USED[sig] = time.time();  
+                LAST_USED[sig] = time.time()
                 return POOL[sig]
             if psutil.virtual_memory().percent >= MEM_LIMIT:
-                raise MemoryError("RAM pressure – new browser denied")
-            crawler = AsyncWebCrawler(config=cfg, thread_safe=False)
+                raise MemoryError("RAM pressure - new browser denied")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)